securedrop/source_app/main.py

```diff
@@ -54,22 +55,30 @@ def generate() -> Union[str, werkzeug.Response]:
         codenames = session.get('codenames', {})
         codenames[tab_id] = codename
         session['codenames'] = fit_codenames_into_cookie(codenames)
+        session["codenames_expire"] = datetime.utcnow() + timedelta(
+            minutes=config.SESSION_EXPIRATION_MINUTES
+        )
```

Comment:
I am not sure why the codenames from the /generate flow need to expire, but this brings the previous behavior back (so that they do expire).

Reply:
I don't know if this was the original motivation or not, but one reason I can see to expire codenames is that folks may have left their terminal on the "generate codename" page, then returned to it at a later point, increasing the risk that a third party observed the codename.